compiler/rustc_trait_selection/src/traits/fulfill.rs

```diff
@@ -247,8 +247,9 @@ struct FulfillProcessor<'a, 'b, 'tcx> {
     register_region_obligations: bool,
 }
 
-fn mk_pending(os: Vec<PredicateObligation<'tcx>>) -> Vec<PendingPredicateObligation<'tcx>> {
+fn mk_pending(os: &[PredicateObligation<'tcx>]) -> Vec<PendingPredicateObligation<'tcx>> {
```

Comment:
I wonder if this is better or worse than the generic `os: impl IntoIterator` version

Reply:
hmm, I personally think that this is fine, we can always change it to be generic once we need to :thinking:
But I also don't think using `impl IntoIterator` is a big problem here so :shrug: